### DISCUSSION TITLE
Loaded object not visible

### DISCUSSION THREAD

**Discussion by RolandZhu** [bulletphysics/bullet3#4001] [Q&A]
Hi! Thanks for all of your support for the great simulator!

I'm working on a manipulation project that requires a robot to interact with 9 objects, one object at a time.
I choose to follow a simple pipeline: load one object, manipulate it, remove it with `removeBody`, and reload the next one.
However, the loaded object becomes invisible after 4 load-reload iterations, while they're an object ID for loaded object and I can get the body information through `getBasePositionAndOrientation`.
I can use the `resetSimulation` to remove everything, but that will require a re-load of the robot, which is not preferred.
Any idea why is this the case and how can I fix it?
I create a minimum example [here](https://www.dropbox.com/s/cp1k27mjul6behu/test.zip?dl=0) to reproduce the issue.
Please let me know if you have any ideas or suggestions!

Many thanks.

**Comment by rohit-kumar-j** [comment]
@RolandZhu 
There must be something wrong with the loading and unloading logic/methods, a little more code would be helpful.

If the robot needs to interact with boxes, see: [createMultiBodyLinks.py](https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/examples/createMultiBodyLinks.py)

If you need to work with indexing, loading, and removing objects from the environment, run [sim_utils.py](https://github.com/rohit-kumar-j/PyBullet_Utilities/blob/main/simulation/sim_utils.py). (However, I have not tested it with explicit URDFs)
You may have to add and subtract a few lines of code in this ⬆️ 

### from sim_utils.py: (Thera are around 20~30 boxes indices in the buffer at a time)

https://user-images.githubusercontent.com/37873142/144284818-9bde1810-b1d4-4176-854c-7dd2a1154c96.mp4

-Rohit
